src/layout/Grid.js

```diff
@@ -58,10 +58,21 @@
         return size;
     };
 
-    Grid.prototype.clearContent = function () {
+    Grid.prototype.clearContent = function (widget) {
         this.content(this.content().filter(function (contentWidget) {
-            contentWidget.target(null);
-            return false;
+            if (!widget) {
+                contentWidget.target(null);
+                return false;
+            }
+            var w = contentWidget;
+            while (w) {
+                if (widget === w) {
```

Comment:
are you sure you want to use this comparison operator here for comparing content widget to widget passed? just checking ...